api/v1alpha1/onload_types.go

```diff
@@ -4,25 +4,76 @@
 package v1alpha1
 
 import (
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// Currently unimplemented
+type OnloadBuildSpec struct {
+}
+
+// Currently unimplemented
+type DevicePluginBuildSpec struct {
+}
+
 // OnloadSpec defines the desired state of Onload
 type OnloadSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// KernelImage
+	// An image that contains the out-of-tree kernel modules used by onload,
+	// including the sfc driver.
+	KernelImage string `json:"kernelImage"`
+
+	// UserImage
+	// An image that contains the built userland objects, used by the cplane
+	// and deviceplugin DaemonSets.
+	UserImage string `json:"userImage"`
+
+	// DevicePluginImage
+	DevicePluginImage string `json:"devicePluginImage"`
+
+	// +optional
+	// Optional version string to associate with this onload CR.
+	Version string `json:"version,omitempty"`
```

Comment:
Do we want to move it to the (existing) `Spec`?

Reply:
This was intended to be the version of onload used by the operator (eg 8.1.0, enterprise 8 etc.)
We could additional information in the top-level `Spec` struct, but this is onload specific.

Reply:
I believe `Version` should be required everywhere for clusters.

Confer context around my proposal:
> Users must manually change the SemVer version to manage an upgrade. The `:latest` tag may still be useful for non-cluster solutions.
>   -- Production Distribution - Draft